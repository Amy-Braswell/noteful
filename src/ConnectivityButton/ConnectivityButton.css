
.Main__button-container {
  max-width: 380px;
  width: 4%;
  margin: 0 10px;
  position: fixed;
  top: 25vh;
  right: 10%;
}

.Main__add-button {
  margin: 0 auto 20px;
}
.Main__add-button:hover {
  background-color: #29862e;
  color: white;
}

.Main__add-button:focus {
  background-color: #29862e;
  color: white;
}

.NavButton {
  background-color: #bbb;
  border: 0;
  outline: 0;
  color: black;
  display: block;
  font-size: 12px;
  font-weight: bold;
  height: 80px;
  padding: 8px;
  text-align: center;
  text-decoration: none;
  width: 120px;
}

a.NavButton {
  padding: 18px 8px 8px;
  line-height:1.3em;
}

.NavButton:hover {
  cursor: pointer;
}

.Nav__back-button {
  max-width: 380px;
  margin: 0 10px;
  position: fixed;
  top: 25vh;
  right: 8.25%;
}

#root > div > main > section > button {
  background-color: #dddddd;
  padding: 5px 15px;
}

#root > div > main > section > button:hover {
  background-color: #29862e;
  color: white;
  }

#root > div > main > section > button > a{
  text-decoration: none;
  color:#0b1f3a;
} 

  