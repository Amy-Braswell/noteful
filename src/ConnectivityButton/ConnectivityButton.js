import React, {Component} from 'react'
import './ConnectivityButton.css'


 class NavConnectivityButton extends Component {

    render(){
    const { tag, className, children, ...otherProps } = this.props

    return React.createElement(
        
        this.props.tag,
        {
            className: ['NavButton', this.props.className].join(' '),
            ...otherProps
        },
        this.props.children

        
    )

    }
}

NavConnectivityButton.defaultProps ={
    tag: 'a'
}

export default NavConnectivityButton