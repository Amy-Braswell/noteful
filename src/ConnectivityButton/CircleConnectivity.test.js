import React from 'react'
import renderer from 'react-test-renderer'
import ConnectivityButton from './ConnectivityButton'
import ReactDOM from 'react-dom'


it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<ConnectivityButton />, div)
  ReactDOM.unmountComponentAtNode(div)
  })

describe(`ConnectivityButton component`, () => {
  const props = {
    tag: 'a',
    className: 'test-class-name',
    children: <p>test children</p>,
    'data-other': 'test-other-prop'
  }

  it('renders a button.ConnectivityButton by default', () => {
    const tree = renderer
      .create(<ConnectivityButton />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders the circle button from props', () => {
    const tree = renderer
      .create(<ConnectivityButton {...props} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
