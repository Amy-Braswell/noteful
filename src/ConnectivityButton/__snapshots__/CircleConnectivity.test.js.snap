// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectivityButton component renders a button.ConnectivityButton by default 1`] = `
<a
  className="NavButton "
/>
`;

exports[`ConnectivityButton component renders the circle button from props 1`] = `
<a
  className="NavButton test-class-name"
  data-other="test-other-prop"
>
  <p>
    test children
  </p>
</a>
`;
